Tasks/VsTest/vstest.ts

```diff
@@ -343,8 +350,249 @@ function getTestResultsDirectory(settingsFile: string, defaultResultsDirectory:
     return defer.promise;
 }
 
+function getTICollectorURI () {
+    return "datacollector://microsoft/TestImpact/1.0";
+}
+
+function getTIAssemblyQualifiedName(vsVersion) {
+    return "Microsoft.VisualStudio.TraceCollector.TestImpactDataCollector, Microsoft.VisualStudio.TraceCollector, Version=" + vsVersion + ".0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a";
+}
+
+function getTIFriendlyName() {
+    return "Test Impact";
+}
+
+function getTestImpactAttributes(vsVersion) {
+    return { uri: getTICollectorURI(),
+             assemblyQualifiedName: getTIAssemblyQualifiedName(vsVersion),
+             friendlyName: getTIFriendlyName() };
+}
+
+function isTestImapctCollectorPresent(dataCollectorArray) {
+    var found = false;
+    var tiaFriendlyName = getTIFriendlyName().toUpperCase();
+    for (var i = 0; i < dataCollectorArray.length; i++) {
+        try {
```

Comment:
why this should throw exception. no function calls - no unexpected exceptions allowed